### PR TITLE
upload last osbs-client.log updates

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- N/A JSON/YAML configuration changes are updated in the relevant schema
- N/A Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 1]
```diff
@@ -376,6 +378,24 @@ def _incremental_upload_logs(self, child_pid=None):
         finally:
             watcher.clean()
 
+    def _upload_logs_once(self):
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/builder_containerbuild.py:381]: I'm trying hard, but the only difference I see here comparing to `_incremental_upload_logs(child_pid=None)` is `except koji.ActionNotAllowed`

Is that exception the only reason to not reuse existing function?

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 2]
```diff
@@ -376,6 +378,13 @@ def _incremental_upload_logs(self, child_pid=None):
...
+        """Upload log updates without waiting for anything"""
+        try:
+            self._incremental_upload_logs()
+        except koji.ActionNotAllowed:
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/builder_containerbuild.py:385]: Maybe I need explanation why. (Shall we log it when this happens, at leas for debugging on koji builder side?)

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 3]
```diff
@@ -533,10 +542,15 @@ def sigint_handler(*args, **kwargs):
...
+            self._upload_logs_once()
             raise ContainerCancelled('Image build was cancelled by OSBS.')
 
         elif not has_succeeded:
-            error_message = self.osbs().get_build_error_message(build_id)
+            error_message = None
+            try:
+                error_message = self.osbs().get_build_error_message(build_id)
+            except Exception as ex:
+                self.logger.error("Error during getting error message: %s", repr(ex))
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/builder_containerbuild.py:553]: Is worth to log it as `logger.exception` to keep traceback for debugging?

File: koji_containerbuild/plugins/builder_containerbuild.py (hunk 2 above)
Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:385]: reopen_file raises that https://github.com/containerbuildsystem/koji-containerbuild/blob/master/koji_containerbuild/plugins/builder_containerbuild.py#L186

when it gets OSError, and there isn't anymore info anyway

I am basically doing the same like https://github.com/containerbuildsystem/koji-containerbuild/blob/master/koji_containerbuild/plugins/builder_containerbuild.py#L505

File: koji_containerbuild/plugins/builder_containerbuild.py (hunk 3 above)
Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:553]: doesn't hurt

File: tests/test_builder_containerbuild.py [hunk 4]
```diff
@@ -715,6 +716,7 @@ def test_createSourceContainer_failure_source(self, reason, expected_exc_type):
         if reason == 'signal_cancelled':
             task._incremental_upload_logs = \
                 lambda pid: os.kill(os.getpid(), signal.SIGINT)
+            flexmock(task).should_receive('_upload_logs_once').once().and_return(True)
```
Comment (MartinBasti) [inline tests/test_builder_containerbuild.py:719]: I'm puzzled here `.and_return(True)` that function doesn't return anything (i.e. `None` in py world)

File: koji_containerbuild/plugins/builder_containerbuild.py (hunk 2 above)
Comment (MartinBasti) [reply koji_containerbuild/plugins/builder_containerbuild.py:385]: ok

File: tests/test_builder_containerbuild.py (hunk 4 above)
Comment (rcerven) [reply tests/test_builder_containerbuild.py]: yeah changed to None, either way it is just to mock that methods to check it was called

**Review by MartinBasti — APPROVED** [review]

**Review by lkolacek — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated: -log logger messages in osbs-client log

**Pull request merged**